jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/startup/CodeWhispererProjectStartupSettingsListener.kt

```diff
@@ -54,6 +54,8 @@ class CodeWhispererProjectStartupSettingsListener(private val project: Project)
     }
 
     override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+        // TODO: investigate why the project level listener not being dispoased when its parent gets disposed
```

Comment:
can you log the full name of the class when this happens to make this easier to investigate?

Reply:
sg